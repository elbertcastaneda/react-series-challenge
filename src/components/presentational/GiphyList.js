import React from 'react';
// import {NavLink} from 'react-router-dom';
import styled from 'styled-components';

const GiphyListContainer = styled.ul`
  list-style: none;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-flow: row wrap;
  justify-content: space-around;

  @media (max-width: 800px) {
    justify-content: space-around;
  }

  @media (max-width: 500px) {
    flex-direction: column;
  }
`;

const GiphyListItem = styled.li`
  display: list-item;
  text-align: -webkit-match-parent;
  width: 31%;
  height: 250px;
  border: 1px solid;
  margin-right: 20px;
  margin-bottom: 20px;
  
  @media (max-width: 1024px) {
    width: 46%;
  }

  @media (max-width: 600px) {
    width: 97%;
  }
`;

const GiphyListItemHeader = styled.h3`
  
  @media (max-width: 600px) {
    
  }
  
`;

const GiphyListPager = styled.ul`
  list-style: none;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-flow: row wrap;
  justify-content: space-around;

  @media (max-width: 800px) {
    justify-content: space-around;
  }

  @media (max-width: 500px) {
    flex-direction: column;
  }
`;

const GiphyListPagerItem = styled.li`
  display: list-item;
  text-align: -webkit-match-parent;
  
  @media (max-width: 1024px) {
    width: 46%;
  }

  @media (max-width: 600px) {
    width: 97%;
  }
`;


// const Link = styled(NavLink)`
//   text-decoration: none;
//   color: white;
//   &:hover {
//     background: #EF9800;
//     outline-width: 0;
//   }

//   &:active {
//     outline-width: 0;
//   }
// `;

const Item = ({source}) => {
  return (
    <GiphyListItem source={source}>
      <header>
        <GiphyListItemHeader>{source.title}</GiphyListItemHeader>
      </header>
    </GiphyListItem>
  );
};

const GiphyListPagerButton = ({label, onClick}) => {
  return (
    <GiphyListPagerItem>
      <button onClick={onClick}>{label}</button>
    </GiphyListPagerItem>
  );
};

const GiphyList = (props) => {
  const { dataSource } = props;

  const renderItems = (data) => {
    return data.map((giphyItem) => {
      return <Item source={giphyItem} key={giphyItem.id} />;
    })
  };
  
  return (
    <GiphyListContainer>
      {renderItems(dataSource.data)}
      <GiphyListPager>
        <GiphyListPagerButton label='First' onClick={props.onClickFirstPage} />
        <GiphyListPagerButton label='Previous' onClick={props.onClickPreviousPage} />
        <GiphyListPagerButton label='Next' onClick={props.onClickNextPage} />
        <GiphyListPagerButton label='Last' onClick={props.onClickLastPage} />
      </GiphyListPager>
    </GiphyListContainer>
  );
};

export default GiphyList;
