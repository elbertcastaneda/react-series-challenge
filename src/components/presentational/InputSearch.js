import React from 'react';
import styled from 'styled-components';

const SectionInput = styled.section`
  position: fixed;
  top: 48px;
  width: 100%;
  height: 40px;
  text-align: center;
  & input {
    padding: 5px;
    text-align: center;
    font-size: 24px;
    color: gray;
    width: 95%;
    height: 100%;
    border: solid 1px lightgrey;
    border-top: 0px;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
  }
`;

const InputSearch = (props) => {
 return (
  <SectionInput>
    <input type='search' {...props} />
  </SectionInput>
 );
};

export default InputSearch;