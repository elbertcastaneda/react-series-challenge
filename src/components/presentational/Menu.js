import React from 'react';
import styled from 'styled-components';

const NAVIGATION = styled.ul`
    list-style: none;
    margin: 0;
    background: #ff9800;
    display: -webkit-box;
    display: -moz-box;
    display: -ms-flexbox;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-flow: row wrap;
    justify-content: flex-end;

    @media (max-width: 500px) {
      flex-direction: column;
    }

    @media (max-width: 800px) {
      justify-content: space-around;
    }
`;

const Menu = (props) => {
  const { children } = props;
  return (
    <NAVIGATION>
      {children}
    </NAVIGATION>
  );
};

export default Menu;
