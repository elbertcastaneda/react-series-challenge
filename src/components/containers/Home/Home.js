import React, { Component } from 'react';
import { HomeContainer } from './Home.styled';
import { GiphyList } from '../../presentational';
import { connect } from 'react-redux';
import trendingGifsActions from '../../../actions/trendingGifs';
const { creators } = trendingGifsActions;
const { getTrendingGifsRequested } = creators;
class Home extends Component {
  componentDidMount () {
    const { getTrendingGifsRequested } = this.props;
    getTrendingGifsRequested();
  }

  render() {
    const {trendingGifs} = this.props;
    return (
      <HomeContainer>
        {/* {JSON.stringify(trendingGifs, null, 2)} */}
        <GiphyList dataSource={trendingGifs} />
      </HomeContainer>
    );
  }
}

const mapStateToProps = (state) => {
  console.log('â€‹mapStateToProps -> state', state);
  const {trendingGifs} = state;

  return {
    trendingGifs
  };
};

export default connect(mapStateToProps, { getTrendingGifsRequested })(Home);
