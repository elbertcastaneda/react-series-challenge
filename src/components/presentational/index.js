import Menu from './Menu';
import ItemMenu from './ItemMenu';
import GiphyList from './GiphyList';
import InputSearch from './InputSearch';

export {
  Menu,
  ItemMenu,
  GiphyList,
  InputSearch
};