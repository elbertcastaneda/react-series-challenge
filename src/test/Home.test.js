import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from '../components/containers/Home';
import {GiphyList} from '../components/presentational';
import store from '../store';
import { Provider } from 'react-redux';
import dataSource from './__mocks/completeFavoritesDataSource';

describe('Home', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Home store={store} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Find controls', () => {
    const wrapper = mount(
      <Provider store={store}><Home trendingGifs={dataSource} /></Provider>
    );
    
    const section = wrapper.find('section');
    expect(section.exists()).toEqual(true);
    const input = wrapper.find('input');
    expect(input.exists()).toEqual(true);
    const value = 'anna faris';
    input.simulate('change', { target: { value } });
    input.simulate('focus');
    expect(wrapper.find(GiphyList)).toHaveLength(1);
  });
});
