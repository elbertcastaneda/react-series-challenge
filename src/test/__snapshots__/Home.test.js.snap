// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<Home
  addFavoriteGifRequested={[Function]}
  getTrendingGifsRequested={[Function]}
  removeFavoriteGifRequested={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  trendingGifs={
    Object {
      "data": Array [],
      "error": "",
      "fetching": false,
      "pagination": Object {
        "count": 0,
        "offset": 1,
        "pages": 0,
        "total_count": 0,
      },
      "query": "",
      "selectedGif": Object {
        "fetching": false,
        "id": "",
        "images": Object {
          "original": Object {
            "url": "",
          },
        },
        "title": "",
      },
      "selectedID": "",
    }
  }
/>
`;
